lib/pbench/test/functional/server/test_user.py

```diff
@@ -0,0 +1,182 @@
+from http import HTTPStatus
+
+import pytest
+from requests import HTTPError
+
+from pbench.client import PbenchServerClient
+
+
+USERNAME1 = "functional"
+USERNAME2 = "nonfunctional"
+PASSWORD = "tests"
+
+
+class TestUser:
+    """
+    Pbench Server functional tests for basic "user" operations.
+
+    NOTE: The functional test system is intended to run on a pristine
+    containerized Pbench Server. We could check for some pre-conditions if we
+    had a known Administrative user (e.g., admin/admin).
+    """
+
+    def test_login_fail(self, pbench_server_client: PbenchServerClient):
+        """
+        Trying to log in to a non-existent user should fail.
+
+        NOTE: This will fail if "nouser" exists.
+        """
+        with pytest.raises(
+            HTTPError, match=r"UNAUTHORIZED for url: http://.*?/api/v1/login"
+        ):
+            pbench_server_client.login("nouser", "nopassword")
+
+    @pytest.mark.parametrize(
+        "json",
+        [
+            {
+                "username": USERNAME1,
+                "first_name": "Test",
+                "last_name": "Account",
+                "password": PASSWORD,
+                "email": f"{USERNAME1}@example.com",
```

Comment:
Are we supposed to use `username` here in format string? or `USERNAME1` works fine as well? Same for line l. 49 and l. 71.

Reply:
It doesn't matter, since we don't expect email. It's possible that as our user model gets more sophisticated we'll want to capture email from Keycloak (e.g., email validation, password change) as part of testing, in which case we'll probably need to create a real `@redhat.com` email address alias for the tests (if we're allowed to do that). Right now ... since we need to provide an email address, I just matched it with the account's username.

Reply:
Got it. I forgot that we have global variable `USERNAME1` and that value is being assigned here.